### ISSUE TITLE
Don't expand history in comments

### ISSUE THREAD

**Issue by jyn514** [oilshell/oil#451]
Current behavior:
```
(osh) ~/.../rust/compiler ▶️ echo hi
hi
(osh) ~/.../rust/compiler ▶️ # !!!
! # echo hi
```
Expected behavior (same as bash):
```
(osh) ~/.../rust/compiler ▶️ echo hi
hi
(osh) ~/.../rust/compiler ▶️ # !!!
(osh) ~/.../rust/compiler ▶️
```

I ran into this when recording a video of my terminal, see the last few seconds of https://asciinema.org/a/Gn7zEZWMeWJluAEM2xRe4jvN0

**Comment by andychu** [comment]
Yes good call... I think we can do this pretty easily but I'll have to look.

The way history is implemented in bash is "interesting"... it's an entirely separate preprocessing/lexing phase that  has special cases to avoid conflicting with constructs like

- `if ! grep`
- `${!x}`
- `[[ !x ]]`

etc.

**Comment by andychu** [comment]
Released with 0.7.pre3: https://www.oilshell.org/release/0.7.pre3/

Thanks for the great bug reports -- keep them coming!
